telemetry/csharp/README.md

```diff
@@ -4,23 +4,32 @@ This solution contains a code generator and datatypes that allow the AWS Toolkit
 
 At this time, the generated code does not function in a standalone capacity. It requires datatypes (within a soon-to-be-published `Amazon.AwsToolkit.Telemetry.Events.Core` namespace). See [Roadmap](#Roadmap) for future plans in this space.
 
+There are two types of telemetry definitions that serve as inputs to the code generator:
+
+-   **Common Telemetry definitions** - Central definitions applied to all AWS Toolkit products. The AWS Toolkit for Visual Studio consumes these telemetry types through a central package (package not generated yet, see Roadmap). These definitions reside in [commonDefinitions.json](/telemetry/definitions/commonDefinitions.json).
+-   **Supplemental Telemetry definitions** - definitions that are specific to one toolkit (or not necessarily applicable to all Toolkits), or definitions that are part of a to-be-released feature which that cannot be made public yet. These definitions reside in the toolkit repo, and the generator is used to produce code that is directly integrated into the toolkit. The generated code references datatypes from the common telemetry definitions, and is expected to have access to the package mentioned above.
```

Comment:
I must admit I don't fully understand what's *different* about the definitions with the exception of where they're generated, my understanding would be that **common** definitions are generated once, in a pipeline off the back of a commit to `aws-toolkit-common` and published to the appropriate package manager - whereas **supplemental** definitions are produced from a model that resides only in the toolkit-specific repo - I'd have thought the mechanics for doing the generation were the same though ... what's the divergence in the logic?

Reply:
Generating code for Supplemental definitions does not generate code for the common definitions. Rather, the generated supplemental code expects to be able to access "already generated" common code.

For example, common telemetry has a type `result` [link](https://github.com/aws/aws-toolkit-common/blob/55b6f443de7793cc659cda1ca59fd073d3fbc342/telemetry/definitions/commonDefinitions.json#L9). I produce a supplemental definition to record when a button is pressed (`button_press`), and track the result, by defining `result` as a metadata field. When running the generator on the supplemental definition, code is not generated for the `result` type, because it exists in the code that was generated for the common definitions.

Reply:
Ahh it does make sense now

Reply:
Ok great -- I've added another paragraph to the file containing the example , to help explain things out